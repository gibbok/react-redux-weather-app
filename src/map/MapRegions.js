// @flow
import * as React from 'react'

/* eslint-disable no-undef */
type PropsType = {
  +regions: Array<{
    +id: string,
    +title:string,
    +isActive:boolean
  }>,
  +onRegionsChange: Function
}
/* eslint-enable no-undef */

const MapRegions:React.StatelessFunctionalComponent<any> = ({ regions, onRegionsChange }:PropsType):React.Element<any> => {
  const options = regions.map((region) =>
    <option
      value={region.id}
      key={region.id}
    >{region.title}
    </option>
  )
  return (
    <select onChange={(e) => onRegionsChange(e.target.value)}>
      {options}
    </select>
  )
}

export default MapRegions
