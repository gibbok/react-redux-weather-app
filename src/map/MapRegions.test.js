import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MapRegions from './MapRegions'

describe('<MapRegions />', () => {
  it('should render', () => {
    const data = {
      regions: [
        {
          id: 'currentLocation',
          title: 'Current Location',
          isActive: true
        },
        {
          id: 'northAmerica',
          title: 'North America',
          isActive: false
        },
        {
          id: 'southAmerica',
          title: 'South America',
          isActive: false
        },
        {
          id: 'europe',
          title: 'Europe',
          isActive: false
        },
        {
          id: 'asia',
          title: 'Asia',
          isActive: false
        },
        {
          id: 'pacificIslands',
          title: 'Pacific Islands',
          isActive: false
        },
        {
          id: 'africa',
          title: 'Africa',
          isActive: false
        }
      ]
    }
    const wrapper = shallow(
      <MapRegions
        regions={data.regions}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
