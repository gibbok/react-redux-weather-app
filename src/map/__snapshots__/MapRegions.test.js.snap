// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapRegions /> should render 1`] = `
<select
  onChange={[Function]}
>
  <option
    value="currentLocation"
  >
    Current Location
  </option>
  <option
    value="northAmerica"
  >
    North America
  </option>
  <option
    value="southAmerica"
  >
    South America
  </option>
  <option
    value="europe"
  >
    Europe
  </option>
  <option
    value="asia"
  >
    Asia
  </option>
  <option
    value="pacificIslands"
  >
    Pacific Islands
  </option>
  <option
    value="africa"
  >
    Africa
  </option>
</select>
`;
